src/search/utils/system.cc

```diff
@@ -46,7 +46,7 @@ bool is_exit_code_error_reentrant(ExitCode exitcode) {
 
 void exit_with(ExitCode exitcode) {
     report_exit_code_reentrant(exitcode);
```

Comment:
It's a drive-by change, but while we're looking at this: can we change this call to a new function `report_exit_code` that plays well with iostreams? At the moment this is likely to break output if we don't flush manually before calling `exit_with`.

Reply:
Done.